src/platform/nodejs/persist/fileSystemPersist.ts

```diff
@@ -0,0 +1,132 @@
+  }
+
+  shift(cb: (err: Error | null, value?: unknown[]) => void): void {
+    this._logger.info('Returning first member of filesystem');
+    this._getFirstFileOnDisk((error, buffer) => {
+      if (error) {
+        cb(error);
+      } else if (buffer) {
+        cb(null, JSON.parse(buffer.toString('utf8')));
+      } else {
+        cb(null);
+      }
+    });
+  }
+
+  /**
+   * Check for temp telemetry files
+   * reads the first file if exist, deletes it and tries to send its load
+   */
+  private _getFirstFileOnDisk(callback: (error: Error | null, value?: Buffer) => void): void {
+    const tempDir = path.join(
+      os.tmpdir(),
+      FileSystemPersist.TEMPDIR_PREFIX + this._options.instrumentationKey,
+    );
+
+    fs.stat(tempDir, (statErr: Error | null, stats: fs.Stats) => {
+      if (stats.isDirectory()) {
+        fs.readdir(tempDir, (error, origFiles) => {
+          if (!error) {
+            const files = origFiles.filter((f) => path.basename(f).includes('.ai.json'));
+            if (files.length === 0) {
+              callback(null);
+            } else {
+              const firstFile = files[0];
+              const filePath = path.join(tempDir, firstFile);
+              fs.readFile(filePath, (readFileErr, payload) => {
+                if (!readFileErr) {
+                  // delete the file first to prevent double sending
+                  fs.unlink(filePath, (unlinkError) => {
+                    if (!unlinkError) {
+                      callback(null, payload);
+                    } else {
+                      callback(unlinkError);
+                    }
+                  });
+                } else {
+                  callback(readFileErr);
+                }
+              });
+            }
+          } else {
+            callback(error);
+          }
+        });
+      }
+    });
+  }
+
+  private _storeToDisk(payload: unknown, cb: (error: Error | null, success?: boolean) => void): void {
+    const directory = path.join(
+      os.tmpdir(),
+      FileSystemPersist.TEMPDIR_PREFIX + this._options.instrumentationKey!,
+    );
+
+    confirmDirExists(directory, (error) => {
+      if (error) {
+        this._logger.warn(`Error while checking/creating directory: `, error && error.message);
+        cb(error);
+        return;
+      }
+
+      getShallowDirectorySize(directory, (err, size) => {
+        if (err || size < 0) {
+          this._logger.warn(`Error while checking directory size: ${err && err.message}`);
+          cb(err);
+          return;
+        }
+        if (size > this.maxBytesOnDisk) {
+          this._logger.warn(
+            `Not saving data due to max size limit being met. Directory size in bytes is: ${size}`,
+          );
+          cb(new Error(`FileSystemPersist size limit reached: ${this.maxBytesOnDisk}`));
+          return;
+        }
+
+        // create file - file name for now is the timestamp, a better approach would be a UUID but that
```

Comment:
What is the problem of having an external dependency here?

Reply:
Most use [`uuid`](https://www.npmjs.com/package/uuid) here